### DISCUSSION TITLE
Bubble up Java Runtime errors on server side

### DISCUSSION THREAD

**Discussion by skhandelwalv** [zeroc-ice/ice#2225] [Q&A]
When a servant thread hits an out of memory error as described here https://doc.zeroc.com/ice/latest/language-mappings/java-mapping/server-side-slice-to-java-mapping/raising-exceptions-in-java the server does not crash. However, the thread is left in the thread pool. This has caused us issue and we would rather fail fast and have ice just restart the server when we hit an OOM is there a way to do that effectively currently?

**Comment by pepone** [comment]
Hi @skhandelwalv,

Thank you for bringing this to our attention. Currently, this feature is not configurable. However, I've created issue #2276 to discuss potential improvements for the next release. Please don't hesitate to comment on the issue with additional thoughts or suggestions.

Best regards,
Jose
